eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/persistence/DbsnpSubmittedVariantAccessioningDatabaseService.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.eva.accession.core.persistence;
+
+import uk.ac.ebi.ampt2d.commons.accession.generators.monotonic.MonotonicRange;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.BasicSpringDataRepositoryDatabaseService;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.service.MonotonicDatabaseService;
+
+import uk.ac.ebi.eva.accession.core.ISubmittedVariant;
+import uk.ac.ebi.eva.accession.core.service.DbsnpSubmittedVariantInactiveService;
+
+import java.util.Collection;
+
+public class DbsnpSubmittedVariantAccessioningDatabaseService
+        extends BasicSpringDataRepositoryDatabaseService<ISubmittedVariant, Long, DbsnpSubmittedVariantEntity>
+        implements MonotonicDatabaseService<ISubmittedVariant, String> {
+
+    public DbsnpSubmittedVariantAccessioningDatabaseService(DbsnpSubmittedVariantAccessioningRepository repository,
+                                                            DbsnpSubmittedVariantInactiveService inactiveService) {
+        super(repository,
+              accessionWrapper -> new DbsnpSubmittedVariantEntity(accessionWrapper.getAccession(),
+                                                                  accessionWrapper.getHash(),
+                                                                  accessionWrapper.getData()),
+              DbsnpSubmittedVariantEntity::getModel,
+              inactiveService);
+    }
+
+    @Override
+    public long[] getAccessionsInRanges(Collection<MonotonicRange> ranges) {
```

Comment:
This is related with a comment I did in other file. We are using an accessioning service in "read only" mode (well, not 100% read only because updates in the accessioned objects are possible). As a consequence of that, we have to implement the "getAccessionsInRanges" method and we don't want to generate new accessions so we have to return an empty array.

Probably create a new service that can query accessions or update accessioned objects is too much just for this. In this case, I would throw an UnsupportedOperationException when calling to this method, to make explicit that it should not be used.

Reply:
agree on throwing unsuported operations wherever applies